### ISSUE TITLE
🐛 Bug Report: Failed to add first workflow in self-hosted mode

### ISSUE THREAD

**Issue by akurdyukov** [notifirehq/notifire#6653]
### 📜 Description

I tried following steps on https://docs.novu.co/community/self-hosting-novu/deploy-with-docker
I cannot add my first workflow, the badge at the top says "You are currently viewing demo workflows. Add your own by completing the setup.".

But I cannot finish the setup because using `npx novu@latest init --secret-key=308286b40fd880b9a7c1771f0a72dfeb` (it's taken from dashboard) failed with `Failed to verify your secret key against https://api.novu.co. For EU instances use --api-url https://eu.novu.co or provide the correct secret key`.

I tried running Local Studio with `npx novu@latest dev --port 4000` which ends with never finding 'Bridge Endpoint scan'. I found out that I need to use command `npx novu@latest dev -o https://deploy-preview-8--onboarding-sandbox.netlify.app` which finds endpoint, but LocalStudio shows black template screen and gives lots of errors in JS console.

### 👟 Reproduction steps

1. Run steps
```
# Get the code
git clone --depth 1 https://github.com/novuhq/novu

# Go to the docker community folder
cd novu/docker/community

# Copy the example env file
cp .env.example .env

# Start Novu
docker-compose -f docker-compose.yml up
```
2. Go to http://localhost:4200/
3. Signup
4. Open "getting started"
5. Execute command from "Initialize a new project" section

### 👍 Expected behavior

LocalStudio should show workflows (I guess)

### 👎 Actual Behavior with Screenshots

<img width="1495" alt="Screenshot 2024-10-08 at 17 41 00" src="https://github.com/user-attachments/assets/6266e2f9-0617-412a-aa76-64a35bd72f4f">

### Novu version

Novu Self Hostet

### npm version

10.7.0

### node version

v20.15.0

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find a similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

**Comment by ayo6706** [comment]
Hello have you being able to fix it?

**Comment by akurdyukov** [comment]
Nope, just stopped any activities on this

**Comment by priyanshdailype** [comment]
you can try using url of api-self-hosted server as api url

**Comment by ArinFaraj** [comment]
The create workflow button is not visible on the current `web` docker image that is used in the `docker-compose.yml` 
But you can still access the workflow creation route by going to http://localhost:4200/workflows/create

**Comment by abbabe** [comment]
I get the error “Failed to verify your secret key against https://api.novu.co. For EU instances use --api-url https://eu.api.novu.co or provide the correct secret key” Is there any other way to solve the problem? How can we remove it from the demo?

**Comment by bastiengrignon** [comment]
> The create workflow button is not visible on the current `web` docker image that is used in the `docker-compose.yml` But you can still access the workflow creation route by going to http://localhost:4200/workflows/create

Thank you for this tips 🙏  but have you been able to promote changes to production ?

**Comment by QusaiKhalid** [comment]
> > The create workflow button is not visible on the current `web` docker image that is used in the `docker-compose.yml` But you can still access the workflow creation route by going to http://localhost:4200/workflows/create
> 
> Thank you for this tips 🙏 but have you been able to promote changes to production ?

Did you find a for promoting workflows from development to production?

**Comment by bastiengrignon** [comment]
> > > The create workflow button is not visible on the current `web` docker image that is used in the `docker-compose.yml` But you can still access the workflow creation route by going to http://localhost:4200/workflows/create
> > 
> > 
> > Thank you for this tips 🙏 but have you been able to promote changes to production ?
> 
> Did you find a for promoting workflows from development to production?

No sorry I've just give up try to migrate from 0.24 to 2.x. I've ended up in their Discord and in a feed someone said that the self-hosted version was still on 0.24 and they planned to release the self-hosted version in April. Until then I won't touch my Novu instance !

**Comment by TNAJanssen** [comment]
> > > The create workflow button is not visible on the current `web` docker image that is used in the `docker-compose.yml` But you can still access the workflow creation route by going to http://localhost:4200/workflows/create
> > 
> > 
> > Thank you for this tips 🙏 but have you been able to promote changes to production ?
> 
> Did you find a for promoting workflows from development to production?

Yeah i think i have! Go to /changes there should be the changes, why the **** did they remove all the buttons and made it so difficult? It was such a great product

**Comment by QusaiKhalid** [comment]
> > > > The create workflow button is not visible on the current `web` docker image that is used in the `docker-compose.yml` But you can still access the workflow creation route by going to http://localhost:4200/workflows/create
> > > 
> > > 
> > > Thank you for this tips 🙏 but have you been able to promote changes to production ?
> > 
> > 
> > Did you find a for promoting workflows from development to production?
> 
> Yeah i think i have! Go to /changes there should be the changes, why the **** did they remove all the buttons and made it so difficult? It was such a great product

Thanks a lot, but just to clarify did the self hosted version have button in previous release? Like did they remove them?!! 
I find it really frustrating that we have to use the url loopholes to crate or promote workflows

**Comment by TNAJanssen** [comment]
> > > > > The create workflow button is not visible on the current `web` docker image that is used in the `docker-compose.yml` But you can still access the workflow creation route by going to http://localhost:4200/workflows/create
> > > > 
> > > > 
> > > > Thank you for this tips 🙏 but have you been able to promote changes to production ?
> > > 
> > > 
> > > Did you find a for promoting workflows from development to production?
> > 
> > Yeah i think i have! Go to /changes there should be the changes, why the **** did they remove all the buttons and made it so difficult? It was such a great product
> 
> Thanks a lot, but just to clarify did the self hosted version have button in previous release? Like did they remove them?!! 
> I find it really frustrating that we have to use the url loopholes to crate or promote workflows

Yeah they did, but i don't know why they removed it...

**Comment by jogiji** [comment]
To add on this, I am trying to implement novu in my nestjs app. Following is my setup
1 : Host Novu Server on coolify instance in my local server exposed to public
2 : Implement Novu framework package in my nestjs app
3 : Create a sample workflow in nestjs code as per documentation
4 : Install Local studio and run it using this command 
`npx novu@latest dev --port 3000 -r /api/v1/novu -d <URL of MY HOSTED NOvu server>`
5 : The studio runs and its able to display the sample workflow


![Image](https://github.com/user-attachments/assets/8b3f7c71-5b2d-439e-85d7-5aa96dcdc237)

6 : But I am not able to create a new workflow from studio, for reasons unknown. I get the below blank window

![Image](https://github.com/user-attachments/assets/703de7e6-97d6-4fb0-8ba5-7cac2d3c62fe)

7 : I am able to sync the locally created workflow to the hosted novu server and it works. But it also deleted the sample workflows that came with default installation.

**Comment by QusaiKhalid** [comment]
@jogiji the studio is just the gui for initializing and inspecting workflows, to create a new workflow you would either need to write the code yourself - just like the sample - or using the local novu dashboard. and regarding the sample workflows and their deletion this the normal behaviour of novu dashboard.

**Comment by jogiji** [comment]
> @jogiji the studio is just the gui for initializing and inspecting workflows, to create a new workflow you would either need to write the code yourself - just like the sample - or using the local novu dashboard. and regarding the sample workflows and their deletion this the normal behaviour of novu dashboard.

Yes, but the local dashboard does not have a create new workflow button.
If you navigate via url then you are able to reach the new workflow definition page.

**Comment by elfico** [comment]
@jogiji 
Refer to this comment: https://github.com/novuhq/novu/issues/6653#issuecomment-2615686808

**Comment by afifurrohman-id** [comment]
Why `/workflows/create` path make it redirect to sample workflow? and didn't show any form for creating new workflow.

even the sample workflow also not working.

![Image](https://github.com/user-attachments/assets/04d8fc54-2695-424a-95e4-9688913c6bd5)

![Image](https://github.com/user-attachments/assets/aec2bbc5-a5d5-42d0-bf7c-a8e77e91118e)

**Comment by scopsy** [comment]
@afifurrohman-id please refer to the latest 2.2 docker images, this web UI is deprecated for self-hosting
